src/Cms/Helpers.php

```diff
@@ -64,29 +64,38 @@ public static function dump($variable, bool $echo = true): string
 	 * @since 3.7.4
 	 *
 	 * @param \Closure $action Any action that may cause an error or warning
-	 * @param \Closure $handler Custom callback like for `set_error_handler()`;
-	 *                          the first argument is a return value override passed
-	 *                          by reference, the additional arguments come from
-	 *                          `set_error_handler()`; returning `false` activates
-	 *                          error handling by Whoops and/or PHP
-	 * @return mixed Return value of the `$action` closure, possibly overridden by `$handler`
+	 * @param \Closure $condition Closure that returns bool to determine if to
+	 *                            suppress an error, receives arguments for
+	 *                            `set_error_handler()`
+	 * @param mixed $fallback Value to return when error is suppressed
+	 * @return mixed Return value of the `$action` closure,
+	 *               possibly overridden by `$fallback`
 	 */
-	public static function handleErrors(Closure $action, Closure $handler)
+	public static function handleErrors(Closure $action, Closure $condition, $fallback = null)
 	{
-		$override = $oldHandler = null;
-		$oldHandler = set_error_handler(function () use (&$override, &$oldHandler, $handler) {
-			$handlerResult = $handler($override, ...func_get_args());
+		$override = null;
 
-			if ($handlerResult === false) {
+		$handler = set_error_handler(function () use (&$override, &$handler, $condition, $fallback) {
+			// check if suppress condition is met
+			$suppress = $condition(...func_get_args());
+
+			if ($suppress === false) {
```

Comment:
I feel like with the new fallback handling, we should rather check for an explicit `true` value. So if the condition returns `null`/`void`, the error won't be suppressed.

```suggestion
			if ($suppress !== true) {
```